presto-geospatial/src/main/java/io/prestosql/plugin/geospatial/GeoFunctions.java

```diff
@@ -1341,6 +1365,33 @@ private static OGCGeometry geomFromBinary(Slice input)
         return geometry;
     }
 
+    private static OGCGeometry geomFromHadoopShape(Slice input)
+    {
+        requireNonNull(input, "input is null");
+
+        byte[] bytes = input.getBytes();
+
+        try {
+            OGCGeometry geometry = OGCGeometry.fromEsriShape(getShapeByteBuffer(bytes));
+            String wkt = GeometryEngine.geometryToWkt(geometry.getEsriGeometry(), getWktExportFlags(bytes));
```

Comment:
Why do we need to roundtrip via `WKT`? Shouldn't it be sufficient to return the result of `OGCGeometry.fromEsriShape(getShapeByteBuffer(bytes))`?

Reply:
The format from the spatial-framework-for-hadoop-package actually serializes the file as (WKID (4 bytes), TYPE (1 byte), ESRI_SHAPE). The method I've written expects this format, as it [drops the first 5 bytes](https://github.com/prestosql/presto/pull/1593/files#diff-7d97c33f8798bc0bcb083a8455c18070R1384) to deserialize just the ESRI shape. I'd be cautious about naming this with ESRI, since we wouldn't normally want to drop those bytes to deserialize only the shape.

The reason I'm doing the roundtrip to WKT is that `OGCGeometry.createFromEsriGeometry` (which `OGCGeometry.fromEsriShape` calls) doesn't handle multi- types properly; empty serialized multilinestrings will be converted into linestrings. The Hadoop package gets around this by also serializing the OGCType. While we can't use this in the OGCGeometry converter, we can use it in the [WKT converter](https://github.com/Esri/geometry-api-java/blob/master/src/main/java/com/esri/core/geometry/GeometryEngine.java#L271) via the export flags.

Reply:
Got it. That makes sense. Let's rename the function to `geometry_from_hadoop_shape`, then.